data.py

```diff
@@ -58,13 +59,153 @@ def jitter_pointcloud(pointcloud, sigma=0.01, clip=0.05):
+class RandomRemoveFourQuadrant:
+    def __call__(self, pointcloud):
+
+        original_length = pointcloud.shape[0]
+        along_which_axis = np.random.randint(3)
+
+        boolean_arr_masked_out = np.ones(original_length)
+        for i in range(3):
+            if i == along_which_axis:
+                continue
+            is_larger = np.random.randint(2)
+            boolean_arr_axis = pointcloud[:, i] > 0 if is_larger else pointcloud[:, i] < 0
+            boolean_arr_masked_out = np.logical_and(boolean_arr_axis, boolean_arr_masked_out)
+
+        masked_pointcloud = pointcloud[np.logical_not(boolean_arr_masked_out), :]
+
+        if len(masked_pointcloud) == 0:
+            return pointcloud
+
+        while masked_pointcloud.shape[0] < original_length:
+            masked_pointcloud = np.concatenate([masked_pointcloud, masked_pointcloud], axis=0)
+
+        masked_pointcloud = masked_pointcloud[:original_length, :]
+        return masked_pointcloud
+
+
+class RandomRemoveTwoQuadrant:
+    def __call__(self, pointcloud):
+        original_length = pointcloud.shape[0]
+        along_which_axis = np.random.randint(3)
+
+        is_larger = np.random.randint(2)
+        boolean_mask = pointcloud[:, along_which_axis] > 0 if is_larger else pointcloud[:, along_which_axis] < 0
+
+        masked_pointcloud = pointcloud[boolean_mask, :]
+
+        if len(masked_pointcloud) == 0:
+            return pointcloud
+
+        while masked_pointcloud.shape[0] < original_length:
+            masked_pointcloud = np.concatenate([masked_pointcloud, masked_pointcloud], axis=0)
+
+        masked_pointcloud = masked_pointcloud[:original_length, :]
+        return masked_pointcloud
+
+
+class RandomRemoveEightQuadrant:
+    """
+    Remove an eighth of the pointcloud that belongs to a quadrant
+    """
+    def __call__(self, pointcloud):
+        original_length = pointcloud.shape[0]
+
+        larger_x = np.random.randint(2)
+        larger_y = np.random.randint(2)
+        larger_z = np.random.randint(2)
+
+        boolean_arr_x = pointcloud[:, 0] > 0 if larger_x else pointcloud[:, 0] < 0
+        boolean_arr_y = pointcloud[:, 1] > 0 if larger_y else pointcloud[:, 1] < 0
+        boolean_arr_z = pointcloud[:, 2] > 0 if larger_z else pointcloud[:, 2] < 0
+
+        masked_out = np.logical_and(boolean_arr_x, boolean_arr_y)
+        masked_out = np.logical_and(masked_out, boolean_arr_z)
+
+        masked_pointcloud = pointcloud[np.logical_not(masked_out), :]
+
+        if len(masked_pointcloud) == 0:
+            return pointcloud
+
+        while masked_pointcloud.shape[0] < original_length:
+            masked_pointcloud = np.concatenate([masked_pointcloud, masked_pointcloud], axis=0)
+
+        masked_pointcloud = masked_pointcloud[:original_length, :]
+
+        return masked_pointcloud
+
+
+class Augment:
+    def __init__(
+            self,
+            transforms: Tuple[Any] = (
+                Jitter(),  # TODO: ALWAYS LEAVE THIS LINE UNCOMMENTED
```

Comment:
ATTENTION IMPORTANT:

This is the modification that you need to make in order to run your experiment